shared-bindings/countio/Counter.c

```diff
@@ -0,0 +1,152 @@
+
+#include <stdint.h>
+
+#include "lib/utils/context_manager_helpers.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "py/runtime0.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/countio/Counter.h"
+#include "shared-bindings/util.h"
+
+//| .. currentmodule:: countio
+//|
+//| :class:`Counter` -- Track the count of edge transistions (pulses) on a given pin
+//| ====================================================================================
+//|
+//| Counter will keep track of the number of edge transistions (pulses) on a given pin
+//|
+//| .. class:: Counter(pin_a)
+//|
+//|   Create an Counter object associated with the given pin. It tracks the number of 
+//|   pulses relative when the object is contructed.
+//|
+//|   :param ~microcontroller.Pin pin_a: Pin to read pulses from.
+//|   
+//|
+//|   For example::
+//|
+//|     import countio
+//|     import time
+//|     from board import *
+//|
+//|     pin_counter = countio.Counter(board.D1)
+//|     
+//|     while True:
+//|         count = pin_counter.count
+//|         if count == None 10:
```

Comment:
What should this be? The `10` looks like it is extra.

Reply:
It was, sorry!. I've updated the example.